processing/src/main/java/org/apache/druid/segment/DimensionHandler.java

```diff
@@ -116,6 +116,7 @@ DimensionMergerV9 makeMerger(
       IndexSpec indexSpec,
       SegmentWriteOutMedium segmentWriteOutMedium,
       ColumnCapabilities capabilities,
+      boolean shouldStore,
```

Comment:
I see what you describe and agree that could be cleaner as we can avoid duplicate code blocks in each dimensionHandler. I'd like to do it as a follow-up when I fix numeric dimensionHandlers if you are OK since currently only the string dimensionHandler checks this flag.